relay-event-normalization/src/normalize/mod.rs

```diff
@@ -343,7 +343,23 @@ impl AiOperationTypeMap {
         if !self.is_enabled() {
             return None;
         }
-        self.operation_types.get(span_op).map(String::as_str)
+
+        // First try exact match by creating a Pattern from the span_op
+        let exact_key = Pattern::new(span_op).ok()?;
+        if self.operation_types.contains_key(&exact_key) {
+            return self.operation_types.get(&exact_key).map(String::as_str);
+        }
```

Comment:
I don't understand this part. Why would we set the type to the _key_ of the hash map instead of the value?